### ISSUE TITLE
Monospaced variant for Inter

### ISSUE THREAD

**Issue by donysukardi** [rsms/inter#128]
It'd be nice if we have a monospaced variant of Inter, something like Roboto Mono and San Francisco Mono. Perhaps even cursive for italic like Operator Mono.

**Comment by originecode** [comment]
I’m not sure a cursive font match Inter design but I agree with you, a monospaced variant of Inter is a really good idea.

**Comment by cihantas** [comment]
If a monospace font is considered will it support ligatures like Fira Code?

**Comment by be5invis** [comment]
There could be two monospace variants:

- Letter width ≈ 0.6em — to use in a western context
- Letter width = 0.5em — to couple with East Asian characters

**Comment by Derpitron** [comment]
In the website for the font (rsms.me/inter), It shows in the bottom that one of the early designs for Inter 'took an almost mono-spaced appearance,  making it really easy to read numbers, punctuation and very short words, but eye-straining to read anything longer.'. After this, Inter was rebooted and made more like what it is today. Maybe that monospace prototype of Inter could be revived for this purpose?

**Comment by orschiro** [comment]
Would be totally amazing and replace Hack as monospace font for me. :-)

**Comment by arronhunt** [comment]
💯 this would be an awesome enhancement and would allow us to fully adopt Inter

**Comment by greg-wolff** [comment]
+1 still got my eye on this, really need mono though

**Comment by axe312ger** [comment]
Came across here about half a year ago. Now setting up my terminal again. Still would love to have a monospaced version.


Can we fund this work effort somehow for @rsms? In case extra bucks would motivate 🙈

**Comment by tazhate** [comment]
will love the mono version!

**Comment by reinhart1010** [comment]
I think designing a condensed style (#71) first might be a great idea before doing so for monospace. 

Condensed geometric font designs (such as [Barlow](https://tribby.com/fonts/barlow)) might be perfect for both variants for legibility reasons.

**Comment by jtrein** [comment]
Do it. Monospaced.

**Comment by xplosionmind** [comment]
> I think designing a condensed style (#71) first might be a great idea before doing so for monospace.

I totally second this.

Since I’m here, thanks for your amazing job, @rsms! Inter is great! I’m using it literally everywhere, including [my website](https://xplosionmind.tk)

**Comment by fabiospampinato** [comment]
Even "just" having monospaced numbers could improve some use cases significantly.

**Comment by nahtnam** [comment]
I found this font: https://www.recursive.design/ Which somehow lets you specify how much monospacing you want. Maybe Inter can do something similar to this?

**Comment by quirico** [comment]
> Even "just" having monospaced numbers could improve some use cases significantly.

@fabiospampinato It's already there: you should set `font-feature-settings: 'tnum';` in your CSS. Other extra features available are [here](https://rsms.me/inter/lab/?antialias=default).

**Comment by fabiospampinato** [comment]
@quirico That's right! 😨 I need to learn how that works then because I'm building a custom font with only a subset of the glyphs in Inter and I guess that broke this feature for me.

**Comment by quirico** [comment]
> @quirico That's right! 😨 I need to learn how that works then because I'm building a custom font with only a subset of the glyphs in Inter and I guess that broke this feature for me.

I don't knwow if you are already using it and if it works for subsetting Inter webfont, but https://github.com/filamentgroup/glyphhanger was a solid solution for me, in the past!

**Comment by WhatsThatItsPat** [comment]
It would be nice to have a non-coding mono space. Basically, I'd like a mono Inter with normal zeros (no inner slash or dot).

**Comment by pinpox** [comment]
> I found this font: https://www.recursive.design/ Which somehow lets you specify how much monospacing you want. Maybe Inter can do something similar to this?

+1 for this if it is possible. I'm looking for a monospaced font that I can pair with inter, would be great to just have a monospace version of inter itself for code

**Comment by aemi-dev** [comment]
> > I found this font: https://www.recursive.design/ Which somehow lets you specify how much monospacing you want. Maybe Inter can do something similar to this?
> 
> +1 for this if it is possible. I'm looking for a monospaced font that I can pair with inter, would be great to just have a monospace version of inter itself for code

Use JetBrains Mono ;)

**Comment by Morpheus2077** [comment]
Inter would fit perfectly into a monospaced version, any news or preview?

**Comment by JoaoRuss0** [comment]
> Inter would fit perfectly into a monospaced version, any news or preview?

Came to ask this

**Comment by IKolpikov** [comment]
+1, monospace would be great!

**Comment by bobby-tesouro** [comment]
+1 !!

**Comment by eraoul** [comment]
+1

**Comment by michael-andreuzza** [comment]
+1

**Comment by pearsche** [comment]
> > > I found this font: https://www.recursive.design/ Which somehow lets you specify how much monospacing you want. Maybe Inter can do something similar to this?
> > 
> > 
> > +1 for this if it is possible. I'm looking for a monospaced font that I can pair with inter, would be great to just have a monospace version of inter itself for code
> 
> Use JetBrains Mono ;)

I think that Roboto Mono fits better currently.

Also, +1

**Comment by mvllow** [comment]
> usually I need monospace version for numbers. Would be awesome to have it even only for numbers.

Tabular numbers may be helpful: https://rsms.me/inter/#features/tnum

**Comment by VideoBSO** [comment]
> > > > I found this font: https://www.recursive.design/ Which somehow lets you specify how much monospacing you want. Maybe Inter can do something similar to this?
> > > 
> > > 
> > > +1 for this if it is possible. I'm looking for a monospaced font that I can pair with inter, would be great to just have a monospace version of inter itself for code
> > 
> > 
> > Use JetBrains Mono ;)
> 
> I think that Roboto Mono fits better currently.
> 
> Also, +1

Personally I find [Monoid](https://github.com/larsenwork/monoid) to be a very good match.

**Comment by liamfenn** [comment]
> font-feature-settings: 'tnum';

This is exactly what I was looking for to fix sizing on a grid when using a dynamic clock element. Thank you so much!

**Comment by xxKeefer** [comment]
another +1 here. i would use the tnums option but i need to render to a canvas element so please Monospace Inter

**Comment by lucafaggianelli** [comment]
> > usually I need monospace version for numbers. Would be awesome to have it even only for numbers.
> 
> Tabular numbers may be helpful: https://rsms.me/inter/#features/tnum

Wow I didn't know a font could have those feats, you opened a new world for me :) thanks!

**Comment by ronasimi** [comment]
Inter pairs really well with Commit Mono...

**Comment by waterkip** [comment]
> Inter pairs really well with Commit Mono...

I agree, not too long ago I started using that font in combination with Inter in my UI.

**Comment by rsms** [comment]
CommitMono works pretty well with Inter indeed, but it's not a perfect match, in particular when it comes to x height. Vertical alignment is also off by a tiny bit (which only affects really large text, so I'd ignore that.)

![lksdidkdl](https://github.com/rsms/inter/assets/47046/e7615077-3651-48d0-be3f-b6f1552ab54d)

**Comment by carlospalol** [comment]
My [CommitMono](https://commitmono.com/) settings, pairing with Inter:

Changes in their "Customize" UI, over defaults:
```
Regular weight 425
Letter spacing -1%
Ligatures <= != === ON
Arrows >-> => ON
Alt. 'g' ON
Alt. 'i' ON
Alt. '6', '9' ON
```

Settings JSON:
```
{"weight":425,"italic":false,"alternates":{"cv01":false,"cv02":true,"cv03":false,"cv04":true,"cv05":false,"cv06":true,"cv07":false,"cv08":false,"cv09":false,"cv10":false,"cv11":false},"features":{"ss01":true,"ss02":true,"ss03":true,"ss04":true,"ss05":true},"letterSpacing":-1,"lineHeight":1,"fontName":"OK"}
```

**Comment by stereomato** [comment]
> My [CommitMono](https://commitmono.com/) settings, pairing with Inter:
> 
> Changes in their "Customize" UI, over defaults:
> 
> ```
> Regular weight 425
> Letter spacing -1%
> Ligatures <= != === ON
> Arrows >-> => ON
> Alt. 'g' ON
> Alt. 'i' ON
> Alt. '6', '9' ON
> ```
> 
> Settings JSON:
> 
> ```
> {"weight":425,"italic":false,"alternates":{"cv01":false,"cv02":true,"cv03":false,"cv04":true,"cv05":false,"cv06":true,"cv07":false,"cv08":false,"cv09":false,"cv10":false,"cv11":false},"features":{"ss01":true,"ss02":true,"ss03":true,"ss04":true,"ss05":true},"letterSpacing":-1,"lineHeight":1,"fontName":"OK"}
> ```

Alt "l" too, no?

**Comment by abhishek-junghare** [comment]
Following fonts share a similar style to Inter -

- [DM Mono](https://fonts.google.com/specimen/DM+Mono)
- [Geist Mono](https://fonts.google.com/specimen/Geist+Mono)
- [Azeret Mono](https://fonts.google.com/specimen/Azeret+Mono)

**Comment by ghost** [comment]
I love Geist Mono!

**Comment by tamdaz** [comment]
There is also Adwaita Mono font, which has a nearly simmilar shape to "SF Mono".

![Image](https://github.com/user-attachments/assets/9c29becf-8fd6-43c5-8491-96fc5a2da64c)

**Comment by dch82** [comment]
> There is also Adwaita Mono font, which has a nearly simmilar shape to "SF Mono".
> 
> ![Image](https://github.com/user-attachments/assets/9c29becf-8fd6-43c5-8491-96fc5a2da64c)

By the way it's just Iosevka modified to look like GNOME's version of Inter.

I'll see if I can configure Iosevka to look the same way.

**Comment by tamdaz** [comment]
> > There is also Adwaita Mono font, which has a nearly simmilar shape to "SF Mono".
> > ![Image](https://github.com/user-attachments/assets/9c29becf-8fd6-43c5-8491-96fc5a2da64c)
> 
> By the way it's just Iosevka modified to look like GNOME's version of Inter.
> 
> I'll see if I can configure Iosevka to look the same way.

Indeed, this is a recent font used for GNOME 48. Also, I know that Iosevka is fully-customizable https://typeof.net/Iosevka/customizer.

**Comment by eeelbrens** [comment]
I just saw this thread and went ahead to create my custom config of Iosevka (just as Adwaita Mono) to replicate the artistics decision made in Inter as far as I could. End result they look pretty much the same save for some subtle changes (Adwaita Mono uses a slimmer 'I', '0' with a dot and tailed 'l', also I do set 500/800 as default weights for monospace instead of the usual 400/700 pair).

Iosevka Custom .toml file for near-exact replication: 
<details>
<summary>Click for code block...</summary>

```
[buildPlans.IosevkaCustom]
family = "Iosevka Custom"
spacing = "normal"
serifs = "sans"
noCvSs = true
exportGlyphNames = false

[buildPlans.IosevkaCustom.variants.design]
one = "base"
two = "straight-neck-serifless"
four = "closed-serifless"
five = "oblique-arched-serifless"
six = "closed-contour"
eight = "two-circles"
nine = "closed-contour"
zero = "tall-slashed"
capital-g = "toothless-rounded-serifless-hooked"
capital-j = "serifless"
capital-m = "flat-bottom-serifless"
capital-q = "crossing"
capital-w = "straight-flat-top-serifless"
f = "flat-hook-serifless-crossbar-at-x-height"
j = "flat-hook-serifed"
r = "hookless-serifed"
t = "flat-hook-short-neck2"
w = "straight-flat-top-serifless"
y = "straight-turn-serifless"
capital-eszet = "rounded-serifless"
long-s = "flat-hook-serifless"
lower-gamma = "straight"
lower-lambda = "tailed-turn"
lower-upsilon = "straight-serifless"
lower-chi = "semi-chancery-straight-serifless"
cyrl-capital-u = "straight-turn-serifless"
cyrl-u = "straight-turn-serifless"
asterisk = "hex-mid"
brace = "straight"
guillemet = "straight"
number-sign = "slanted"
at = "fourfold"
percent = "rings-continuous-slash"
lig-equal-chain = "without-notch"

[buildPlans.IosevkaCustom.variants.oblique]
a = "single-storey-serifless"
e = "rounded"
f = "flat-hook-tailed-crossbar-at-x-height"
cyrl-a = "single-storey-serifless"
number-sign = "upright"

[buildPlans.IosevkaCustom.weights.Regular]
shape = 500
menu = 400
css = 400

[buildPlans.IosevkaCustom.weights.Bold]
shape = 800
menu = 700
css = 700

[buildPlans.IosevkaCustom.widths.Normal]
shape = 658
menu = 5
css = "normal"

[buildPlans.IosevkaCustom.slopes.Upright]
angle = 0
shape = "upright"
menu = "upright"
css = "normal"

[buildPlans.IosevkaCustom.slopes.Oblique]
angle = 9.4
shape = "oblique"
menu = "italic"
css = "italic"
```
</details>

> I do reserve a few personal preferences in my actual config file which aren't necessary here, replicating Inter's cv02, cv03, cv04, cv05 (yeah I do bring back the tailed 'l' from Adwaita Mono for myself xD), and cv09 for Iosevka too!

@dch82 guess I saved u some effort.

**Edit:** Expanded width a bit from 600 to 658
**Edit2:** Removed all the redundant specifications in italic override, and modified bold to 800 weight instead of 700

**Comment by dantezhu** [comment]
hi, is there any progress?

**Comment by dch82** [comment]
> I just saw this thread and went ahead to create my custom config of Iosevka (just as Adwaita Mono) to replicate the artistics decision made in Inter as far as I could. End result they look pretty much the same save for some subtle changes (Adwaita Mono uses a slimmer 'I', '0' with a dot and tailed 'l', also I do set 500/800 as default weights for monospace instead of the usual 400/700 pair).
> 
> Iosevka Custom .toml file for near-exact replication:
> Click for code block...
> 
> > I do reserve a few personal preferences in my actual config file which aren't necessary here, replicating Inter's cv02, cv03, cv04, cv05 (yeah I do bring back the tailed 'l' from Adwaita Mono for myself xD), and cv09 for Iosevka too!
> 
> [@dch82](https://github.com/dch82) guess I saved u some effort.
> 
> **Edit:** Expanded width a bit from 600 to 658 **Edit2:** Removed all the redundant specifications in italic override, and modified bold to 800 weight instead of 700

Thanks, also I put it into a fork of Iosevka so we can get some more stuff done

https://github.com/dch82/Intersevka

**Comment by haidar47x** [comment]
I'd ditch Source Code Pro for Inter Mono.

**Comment by aretrace** [comment]
@rsms <sub><sup><sub><sup>plz</sup></sub></sup></sub> <sub><sup><sub><sup>😢</sup></sub></sup></sub>

**Comment by umbrageodotus** [comment]
> I just saw this thread and went ahead to create my custom config of Iosevka (just as Adwaita Mono) to replicate the artistics decision made in Inter as far as I could. End result they look pretty much the same save for some subtle changes (Adwaita Mono uses a slimmer 'I', '0' with a dot and tailed 'l', also I do set 500/800 as default weights for monospace instead of the usual 400/700 pair).
> 
> Iosevka Custom .toml file for near-exact replication:
> 
> Click for code block...
> > I do reserve a few personal preferences in my actual config file which aren't necessary here, replicating Inter's cv02, cv03, cv04, cv05 (yeah I do bring back the tailed 'l' from Adwaita Mono for myself xD), and cv09 for Iosevka too!
> 
> [@dch82](https://github.com/dch82) guess I saved u some effort.
> 
> **Edit:** Expanded width a bit from 600 to 658 **Edit2:** Removed all the redundant specifications in italic override, and modified bold to 800 weight instead of 700

Why base it on Iosevka when commit mono with the settings mentioned here is so much closer?

**Comment by dch82** [comment]
> > I just saw this thread and went ahead to create my custom config of Iosevka (just as Adwaita Mono) to replicate the artistics decision made in Inter as far as I could. End result they look pretty much the same save for some subtle changes (Adwaita Mono uses a slimmer 'I', '0' with a dot and tailed 'l', also I do set 500/800 as default weights for monospace instead of the usual 400/700 pair).
> > 
> > Iosevka Custom .toml file for near-exact replication:
> > 
> > Click for code block...
> > > I do reserve a few personal preferences in my actual config file which aren't necessary here, replicating Inter's cv02, cv03, cv04, cv05 (yeah I do bring back the tailed 'l' from Adwaita Mono for myself xD), and cv09 for Iosevka too!
> > 
> > [@dch82](https://github.com/dch82) guess I saved u some effort.
> > 
> > **Edit:** Expanded width a bit from 600 to 658 **Edit2:** Removed all the redundant specifications in italic override, and modified bold to 800 weight instead of 700
> 
> Why base it on Iosevka when commit mono with the settings mentioned here is so much closer?

Today was the first time I've heard of it, I guess.
Edit: just seen the above mentioned

**Comment by umbrageodotus** [comment]
@dch82 Try it! It's a better fit for a starting point IMO. Use the settings mentioned above in this issue.

**Comment by eeelbrens** [comment]
> > I just saw this thread and went ahead to create my custom config of Iosevka (just as Adwaita Mono) to replicate the artistics decision made in Inter as far as I could. End result they look pretty much the same save for some subtle changes (Adwaita Mono uses a slimmer 'I', '0' with a dot and tailed 'l', also I do set 500/800 as default weights for monospace instead of the usual 400/700 pair).
> > Iosevka Custom .toml file for near-exact replication:
> > Click for code block...
> > > I do reserve a few personal preferences in my actual config file which aren't necessary here, replicating Inter's cv02, cv03, cv04, cv05 (yeah I do bring back the tailed 'l' from Adwaita Mono for myself xD), and cv09 for Iosevka too!
> > 
> > 
> > [@dch82](https://github.com/dch82) guess I saved u some effort.
> > **Edit:** Expanded width a bit from 600 to 658 **Edit2:** Removed all the redundant specifications in italic override, and modified bold to 800 weight instead of 700
> 
> Why base it on Iosevka when commit mono with the settings mentioned here is so much closer?

@umbrageodotus @dch82 Just revisitied this thread and here's my own take. Commit Mono is rather dreadful to use (has a lot of Inconsolata-like quirks which prevents it from being a faithful Inter Mono or even a good terminal font imo) and dealing with Commit's options upon download is much messier and primitive than toggling a font's cvs (or Iosevka's custom builds).
Also, if you're aiming for an actual font pairing, Iosevka would give you the flexibility you need to give your monospace font the distinguishing features you need in a terminal for example.
I actually switched from my Iosevka's build to my beloved Jetbrains Mono (with cv10, cv11, cv12) a while ago to save my eyes, though am thinking of switching back for a more conservative width 500 build.

**Comment by dch82** [comment]
Only problem with iosevka I see is lack of variable weight.

**Comment by eeelbrens** [comment]
@dch82 not that monospace fonts need variable weights in terminals anyways, but would sure help
also, it seems like [too much work for them to implement](https://github.com/be5invis/Iosevka/issues/300)

**Comment by dch82** [comment]
@eeelbrens true, but a lot of people love Inter partly due to its low weight as a variable font.

I had to scrap Iosevka on my website because it was *ever so heavy* unless you take the time to trim it down!

**Comment by eeelbrens** [comment]
@dch82 well guessed you'd complain for website hosting or smth ofc variable fonts would be superior in that regard
ok so that's one more checklist item for rsms if he decided to actually make Inter Mono lol

**Comment by oxidebits** [comment]
I love the Inter font, but I’m longing for a monospace sibling. Rasmus, would you consider creating an Inter Mono?

In return, you will receive the eternal blessings of developers, granting you immortality.

**Comment by 40a763** [comment]
I think [Paper Mono](https://github.com/paper-design/paper-mono) pairs very well with Inter.

<img width="877" height="558" alt="Image" src="https://github.com/user-attachments/assets/368845aa-b752-466a-9627-d7d293dc44b4" />

**Comment by slahapman** [comment]
Came across this in searching for fonts that look close to SF Mono/pair well with Inter. Not sure if anyone else has found this, but it seems pretty close. 

https://github.com/lujstn/inter-mono

EDIT: Testing it out myself for a little while, I concluded it wasn't for me. Ended up just sticking with Noto Sans Mono (what I was using prior).

EDIT 2: Ended up checking out Paper Mono, and aside from patching it to force a slashed 0, I think it's more or less perfect for what I'm looking for. I did my own customized Iosevka, Commit Mono, and a few others, but I really think paper is what I'm looking for.

**Comment by santos7814-arch** [comment]
Inter Mono 18 Style Download

**Comment by santos7814-arch** [comment]
New Inter Mono 4.2 by @rsms 

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/8b54f1dc-9358-4d52-9f51-ec2e1f755456" />

**Comment by xls2009** [comment]
I recently made a monospaced font to match well with Inter font, called Sindy Mono:
<img width="1200" height="900" alt="Image" src="https://github.com/user-attachments/assets/aa9d82ed-6402-4aa0-96e4-0ed644bfe868" />
I'm not really sure if this is related or not, but you can download this font at my GitHub repository.
[Download the Sindy Mono font](https://github.com/xls2009/Sindy-Mono-Font-Family)
